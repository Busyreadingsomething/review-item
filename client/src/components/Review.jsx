import React from 'react';
import ReviewEntry from './ReviewEntry';
import itemData from '../../../dummyData';

class Review extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      data: itemData[2].reviews,
    };
  }
  render() {
    return (
      <div>{this.state.data.map(item => <ReviewEntry key={item.username} data={item} />)}
      </div>
    );
  }
}

export default Review;
