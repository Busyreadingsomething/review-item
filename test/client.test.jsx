import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Review from '../client/src/components/Review';

describe('Review component', () => {
  it('should be defined', () => {
    expect(Review).toBeDefined();
  });
  it('should render correctly', () => {
    const tree = renderer.create(<Review />);
    expect(tree).toMatchSnapshot();
  });
});
