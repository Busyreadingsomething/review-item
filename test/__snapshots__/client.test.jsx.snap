// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review component should render correctly 1`] = `
<div>
  <table
    id="table"
  >
    <tbody
      id="tbody"
    >
      <tr>
        <td
          id="username"
        >
          abartoletti8
        </td>
      </tr>
      <tr>
        <td
          id="star"
        >
          5
           star: 
        </td>
        <td
          id="reviewTitle"
        >
          great
        </td>
      </tr>
      <tr>
        <td
          colSpan="2"
          id="data"
        >
          4/10/2017
        </td>
      </tr>
      <tr>
        <td
          colSpan="2"
          id="purchased"
        >
          
           
        </td>
      </tr>
      <tr>
        <td
          colSpan="2"
        >
          This product is wonderful. best product ever!
           
        </td>
      </tr>
    </tbody>
  </table>
  <table
    id="table"
  >
    <tbody
      id="tbody"
    >
      <tr>
        <td
          id="username"
        >
          wdalby7
        </td>
      </tr>
      <tr>
        <td
          id="star"
        >
          5
           star: 
        </td>
        <td
          id="reviewTitle"
        >
          wonderful
        </td>
      </tr>
      <tr>
        <td
          colSpan="2"
          id="data"
        >
          11/11/2017
        </td>
      </tr>
      <tr>
        <td
          colSpan="2"
          id="purchased"
        >
          Verified Purchase
           
        </td>
      </tr>
      <tr>
        <td
          colSpan="2"
        >
          This product is tony. its greeeeeeaaaat!
           
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
